static/js/components/dashboard/courses/ProgressMessage.js

```diff
@@ -0,0 +1,174 @@
+// @flow
+/* global SETTINGS: false */
+import React from 'react';
+import R from 'ramda';
+import _ from 'lodash';
+import moment from 'moment';
+
+import type { CourseRun } from '../../../flow/programTypes';
+import {
+  STATUS_PASSED,
+  STATUS_NOT_PASSED,
+  STATUS_CAN_UPGRADE,
+  STATUS_MISSED_DEADLINE,
+  STATUS_CURRENTLY_ENROLLED,
+  STATUS_OFFERED,
+  STATUS_WILL_ATTEND,
+  STATUS_PENDING_ENROLLMENT,
+  DASHBOARD_FORMAT,
+  EDX_LINK_BASE,
+} from '../../../constants';
+import { renderSeparatedComponents } from '../../../util/util';
+import { ifValidDate } from '../../../util/date';
+import { hasAnyStaffRole } from '../../../lib/roles';
+import Progress from './Progress';
+import { courseStartDateMessage } from './util';
+
+// ProgressMessage is ONLY displayed for users who are already enrolled
+
+const courseHasStarted = (courseRun: CourseRun): boolean => (
+  moment(courseRun.course_start_date).isBefore(moment())
+);
+
+const courseMessage = (courseRun: CourseRun) => (
+  courseHasStarted(courseRun) ? "Course in progress" : courseStartDateMessage(courseRun)
+);
+
+export default class ProgressMessage extends React.Component {
+  props: {
+    courseRun:               CourseRun,
+    openCourseContactDialog: () => void,
+  };
+
+  renderCourseDateMessage(label: string, dateString: string): string {
+    let date = moment(dateString);
+    return ifValidDate('', date => `${label}: ${date.format(DASHBOARD_FORMAT)}`, date);
+  }
+
+  renderStartDateMessage(run: CourseRun, shouldStartInFuture: boolean): string|null {
+    if (run.course_start_date) {
+      return this.renderCourseDateMessage('Start date', run.course_start_date);
+    } else if (run.fuzzy_start_date && shouldStartInFuture) {
+      return `Coming ${run.fuzzy_start_date}`;
+    } else {
+      return null;
+    }
+  }
+
+  renderDetailContents(run: CourseRun) {
```

Comment:
no, I haven't yet gone through to delete unused code, so there is probably a lot of stuff which will be in that state.